### PR TITLE
Display tips when starting and ending the download

### PR DESCRIPTION
**Pull request by sunyuzhao**

### CODE REVIEW

File: src/message-poller.cpp [hunk 1]
```diff
@@ -55,6 +55,15 @@ class SeaDriveEvent {
         } else {
             qWarning("unknown type of seadrive event %s", toCStr(type));
             event.fs_op_error = UNKNOWN_ERROR;
+            if(type == "file-download.start") {
+                QString title = QObject::tr("Download file");
+                QString msg = QObject::tr("Start to download file \"%1\" ").arg(::getBaseName(json.getString("path")));
+                gui->trayIcon()->showWarningMessage(title, msg);
+            } else if (type == "file-download.done") {
+                QString title = QObject::tr("Download file");
+                QString msg = QObject::tr("Succeed to download file \"%1\" ").arg(::getBaseName(json.getString("path")));
```
Comment (freeplant) [inline src/message-poller.cpp]: file \"%1\" has been downloaded.

File: src/message-poller.cpp [hunk 2]
```diff
@@ -235,6 +236,18 @@ void MessagePoller::processNotification(const SyncNotification& notification)
...
+        QString title = tr("Download file");
+        QString msg = tr("Start to download file \"%1\" ").arg(::getBaseName(event.path));
+        gui->trayIcon()->showMessage(title, msg, "NULL");
+    } else if (event.type == "file-download.done") {
+        QString title = tr("Download file");
+        QString msg = tr("file \"%1\" has been downloaded ").arg(::getBaseName(event.path));
+        gui->trayIcon()->showMessage(title, msg, "NULL");
+    }
+    type_ = event.type;
+    path_ = event.path;
```
Comment (lins05) [inline src/message-poller.cpp]: 这里应该 return 了

File: src/ui/transfer-progress-dialog.h [hunk 3]
```diff
@@ -19,6 +19,7 @@ class TransferProgressDialog : public QDialog
 public:
     TransferProgressDialog(QWidget *parent = 0);
     void resizeEvent(QResizeEvent* event) Q_DECL_OVERRIDE;
+    void setDownloadUi() { tab_widget_->setCurrentIndex(DOWNLOAD); };
```
Comment (lins05) [inline src/ui/transfer-progress-dialog.h]: `setDownloadUi` => `showDownloadTab`

File: src/message-poller.h [hunk 4]
```diff
@@ -45,6 +45,8 @@ class MessagePoller : public QObject {
     ~MessagePoller();
 
     void start();
+    QString eventType() const { return type_; }
```
Comment (lins05) [inline src/message-poller.h]: `eventType` => `lastEventType`

File: src/message-poller.cpp [hunk 5]
```diff
@@ -235,6 +236,18 @@ void MessagePoller::processNotification(const SyncNotification& notification)
 
 void MessagePoller::processSeaDriveEvent(const SeaDriveEvent &event)
 {
+    if(event.type == "file-download.start") {
+        QString title = tr("Download file");
+        QString msg = tr("Start to download file \"%1\" ").arg(::getBaseName(event.path));
+        gui->trayIcon()->showMessage(title, msg, "NULL");
```
Comment (lins05) [inline src/message-poller.cpp]: "NULL" 去掉

File: src/message-poller.cpp [hunk 6]
```diff
@@ -235,6 +236,18 @@ void MessagePoller::processNotification(const SyncNotification& notification)
...
+    } else if (event.type == "file-download.done") {
+        QString title = tr("Download file");
+        QString msg = tr("file \"%1\" has been downloaded ").arg(::getBaseName(event.path));
+        gui->trayIcon()->showMessage(title, msg, "NULL");
```
Comment (lins05) [inline src/message-poller.cpp]: 一样， "NULL" 去掉

File: src/message-poller.h [hunk 7]
```diff
@@ -45,6 +45,8 @@ class MessagePoller : public QObject {
...
+    QString eventPath() const { return path_; }
```
Comment (lins05) [inline src/message-poller.h]: lastEventPath

File: src/message-poller.h [hunk 8]
```diff
@@ -66,6 +68,8 @@ private slots:
     SeafileRpcClient *rpc_client_;
 
     QTimer *check_notification_timer_;
+    QString type_;
+    QString path_;
```
Comment (lins05) [inline src/message-poller.h]: last_event_type_, last_event_path_

File: src/ui/tray-icon.cpp [hunk 9]
```diff
@@ -644,6 +645,13 @@ void SeafileTrayIcon::onMessageClicked()
 
     // DiffReader *reader = new DiffReader(repo, previous_commit_id_, commit_id_);
     // QThreadPool::globalInstance()->start(reader);
+    if (gui->messagePoller()->eventType() == "file-download.start") {
```
Comment (lins05) [inline src/ui/tray-icon.cpp]: 把上面的 repo_id_.isEmpty 的判断注释掉

File: src/ui/tray-icon.h [hunk 10]
```diff
@@ -18,6 +18,7 @@ class LoginDialog;
 class TrayNotificationManager;
 class SyncErrorsDialog;
 class TransferProgressDialog;
+class MessagePoller;
```
Comment (lins05) [inline src/ui/tray-icon.h]: 这一行删掉

**Pull request merged**